packages/exec/src/exec.ts

```diff
@@ -28,3 +28,44 @@ export async function exec(
   const runner: tr.ToolRunner = new tr.ToolRunner(toolPath, args, options)
   return runner.exec()
 }
+
+export async function getExecOutput(
+  commandLine: string,
+  args?: string[],
+  options?: ExecOptions
+): Promise<ExecOutput> {
+  let stdout = ''
+  let stderr = ''
+
+  const originalStdoutListener = options?.listeners?.stdout
+  const originalStdErrListener = options?.listeners?.stderr
+
+  const stdErrListener = (data: Buffer): void => {
+    stderr += data.toString()
+    if (originalStdErrListener) {
+      originalStdErrListener(data)
+    }
+  }
+
+  const stdOutListener = (data: Buffer): void => {
+    stdout += data.toString()
+    if (originalStdoutListener) {
+      originalStdoutListener(data)
+    }
+  }
+
+  const listeners: ExecListeners = {
+    ...options?.listeners,
+    stdout: stdOutListener,
+    stderr: stdErrListener
+  }
+
+  const exitCode = await exec(commandLine, args, {...options, listeners})
+
+  //return undefined for stdout/stderr if they are empty
```

Comment:
Why undefined over the empty string? If there's no output, returning "" should be fine